BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1874 @@
++            <artifactId>adventure-api</artifactId>
++        </dependency>
++        <dependency>
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-text-serializer-gson</artifactId>
++        </dependency>
++        <dependency>
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-text-serializer-legacy</artifactId>
++        </dependency>
++        <dependency>
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-text-serializer-bungeecord</artifactId>
++            <version>4.0.0-SNAPSHOT</version>
++        </dependency>
+     </dependencies>
+ </project>
+diff --git a/api/src/main/java/net/md_5/bungee/api/CommandSender.java b/api/src/main/java/net/md_5/bungee/api/CommandSender.java
+index d5888311..a8224a9f 100644
+--- a/api/src/main/java/net/md_5/bungee/api/CommandSender.java
++++ b/api/src/main/java/net/md_5/bungee/api/CommandSender.java
+@@ -1,9 +1,13 @@
+ package net.md_5.bungee.api;
+ 
+ import java.util.Collection;
++// Waterfall start - we're going on an adventure!
++import net.kyori.adventure.audience.Audience;
++import net.kyori.adventure.text.Component;
++// Waterfall end
+ import net.md_5.bungee.api.chat.BaseComponent;
+ 
+-public interface CommandSender
++public interface CommandSender extends Audience // Waterfall - we're going on an adventure!
+ {
+ 
+     /**
+@@ -34,14 +38,18 @@ public interface CommandSender
+      * Send a message to this sender.
+      *
+      * @param message the message to send
++     * @deprecated in favour of {@link Audience#sendMessage(Component)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
+     public void sendMessage(BaseComponent... message);
+ 
+     /**
+      * Send a message to this sender.
+      *
+      * @param message the message to send
++     * @deprecated in favour of {@link Audience#sendMessage(Component)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
+     public void sendMessage(BaseComponent message);
+ 
+     /**
+diff --git a/api/src/main/java/net/md_5/bungee/api/ProxyServer.java b/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
+index a4011335..f20f401b 100644
+--- a/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
++++ b/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
+@@ -4,11 +4,19 @@ import com.google.common.base.Preconditions;
+ import java.io.File;
+ import java.net.InetSocketAddress;
+ import java.net.SocketAddress;
++import java.text.MessageFormat;
+ import java.util.Collection;
++import java.util.Locale;
+ import java.util.Map;
+ import java.util.UUID;
+ import java.util.logging.Logger;
+ import lombok.Getter;
++// Waterfall start - we're going on an adventure!
++import net.kyori.adventure.audience.Audience;
++import net.kyori.adventure.audience.ForwardingAudience;
++import net.kyori.adventure.text.Component;
++// Waterfall end
++import net.kyori.adventure.text.TranslatableComponent;
```

Comment:
move these into their own import block at the end

Reply:
Sir yes sir xD